src/server/script_mgr.cc

```diff
@@ -174,4 +179,66 @@ void ScriptMgr::PrintLatency(ConnectionContext* cntx) {
   }
 }
 
+bool ScriptMgr::Insert(std::string_view id, std::string_view body) {
+  bool updated = false;
+
+  lock_guard lk{mu_};
+  auto [it, _] = db_.emplace(id, InternalScriptData{default_params_, nullptr});
+
+  if (auto& body_ptr = it->second.body; !body_ptr) {
+    updated = true;
+
+    body_ptr.reset(new char[body.size() + 1]);
+    memcpy(body_ptr.get(), body.data(), body.size());
+    body_ptr[body.size()] = '\0';
+  }
+
+  UpdateScriptCaches(id, it->second);
+
+  return updated;
+}
+
+optional<ScriptMgr::ScriptData> ScriptMgr::Find(std::string_view sha) const {
+  if (sha.size() != ScriptKey{}.size())
+    return std::nullopt;
+
+  lock_guard lk{mu_};
+  if (auto it = db_.find(sha); it != db_.end() && it->second.body)
+    return ScriptData{it->second, it->second.body.get()};
+
+  return std::nullopt;
+}
+
+vector<pair<string, string>> ScriptMgr::GetAll() const {
+  vector<pair<string, string>> res;
+
+  lock_guard lk{mu_};
+  res.reserve(db_.size());
+  for (const auto& [sha, data] : db_) {
+    res.emplace_back(string{sha.data(), sha.size()}, data.body.get());
+  }
+
+  return res;
+}
+
+void ScriptMgr::UpdateScriptCaches(ScriptKey sha, ScriptParams params) const {
+  shard_set->pool()->Await([&sha, &params](auto index, auto* pb) {
+    ServerState::tlocal()->SetScriptParams(sha, params);
+  });
+}
+
+bool ScriptMgr::ScriptParams::ApplyPragma(string_view pragma, ScriptParams* params) {
```

Comment:
add a comment here showing how the pragma would look like in a script.

Reply:
We don't parse pragmas from scripts yet